pkgs/development/libraries/flutter-engine/package.nix

```diff
@@ -0,0 +1,177 @@
+      url = "https://commondatastorage.googleapis.com/chrome-linux-sysroot/toolchain/79a7783607a69b6f439add567eb6fcb48877085c/debian_sid_${toolchainArch}_sysroot.tar.xz";
+      sha256 = "sha256-+jooJ4YcWHUfD5AT+Yhe0nYQNBl7IvJl2dRc+d3Yl+g=";
+    };
+
+    dontUnpack = true;
+    dontConfigure = true;
+    dontBuild = true;
+
+    ## FIXME: use "flutter-deps" and the arguments which the OG script takes to generate the git revision
+    gitrev = writeText "git-revision.py" ''
+      #!${python3}/bin/python3
+
+      print("${fakeHash}")
+    '';
+
+    dartPackageConfig = writeText "dart-package-config.json" ''
+      {
+        "configVersion": 2,
+        "generator": "tools/generate_package_config.dart",
+        "packages": []
+      }
+    '';
+
+    binaryFixes = [
+      "src/third_party/dart/tools/sdks/dart-sdk/bin/dart"
+      "src/flutter/third_party/gn/gn"
+    ];
+
+    shellFixes = [
+      "src/third_party/dart/tools/sdks/dart-sdk/bin/dart2js"
+      "src/flutter/tools/gn"
+    ];
+
+    ## FIXME: instead of patching a bunch of binaries, maybe we copy from nixpkgs?
+    installPhase = ''
+      mkdir -p $out
+      cp -r -P --no-preserve=mode,ownership $buildroot $out/src
+      cp -r -P --no-preserve=mode,ownership $src $out/src/flutter
+
+      ${concatStringsSep "\n" (attrValues (mapAttrs (name: src: ''
+        if [[ -d ${src} ]]; then
+          mkdir -p $out/${name}
+          cp -r -P --no-preserve=mode,ownership ${src}/* $out/${name}
+          find ${src} -type f -name '.*' | xargs -I {} cp --no-preserve=mode,ownership {} $out/${name}
+        else
+          cp -P --no-preserve=mode,ownership ${src} $out/${name}
+        fi
+      '') flutter-deps))}
+
+      touch $out/src/third_party/dart/build/config/gclient_args.gni
+      cp $gitrev $out/src/flutter/build/git_revision.py
+
+      for bin in $binaryFixes; do
+        chmod 0755 $out/$bin
+        patchelf --set-interpreter ${stdenv.cc.libc}/lib/${interpreter} $out/$bin
+      done
+
+      for sh in $shellFixes; do
+        chmod 0755 $out/$sh
+        patchShebangs $out/$sh
+      done
+
+      chmod +x $out/src/build/linux/sysroot_ld_path.sh
+
+      sed -i '1 s|^.*$|#!${gnumake}/bin/make -f|' $out/src/third_party/harfbuzz/src/update-unicode-tables.make
+
+      mkdir -p $out/src/build/linux/debian_sid_${toolchainArch}-sysroot
+      tar -xf $toolchain -C $out/src/build/linux/debian_sid_${toolchainArch}-sysroot
+
+      cd $out
+      mkdir -p $out/src/third_party/dart/.dart_tool
+      cp -r --no-preserve=ownership,mode $dartPackageConfig $out/src/third_party/dart/.dart_tool/package_config.json
+
+      python3 $out/src/third_party/dart/tools/generate_package_config.py
+      python3 $out/src/third_party/dart/tools/generate_sdk_version_file.py
+      python3 $out/src/tools/remove_stale_pyc_files.py src/tools
+      python3 $out/src/flutter/tools/pub_get_offline.py
+    '';
+  };
+in stdenvNoCC.mkDerivation rec {
+  pname = "flutter-engine-${runtimeMode}";
```

Comment:
seems wasteful to have separate builds for each runtimeMode

can we have one core build and multiple lightweight wrappers?

Reply:
It's because we'll have to produce *x* number of configured builds. Imo, it's more efficient doing it this way.